extensions/amp-embedly-card/1.0/test/test-amp-embedly-card.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright 2021 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import '../amp-embedly-card';
+import {createElementWithAttributes} from '#core/dom';
+import {doNotLoadExternalResourcesInTest} from '#testing/iframe';
+import {toggleExperiment} from '#experiments';
+import {waitFor} from '#testing/test-helper';
+
+describes.realWin(
+  'amp-embedly-card-v1.0',
+  {
+    amp: {
+      extensions: ['amp-embedly-card:1.0'],
+    },
+  },
+  (env) => {
+    let win;
+    let doc;
+    let element;
+
+    beforeEach(async () => {
+      win = env.win;
+      doc = win.document;
+      toggleExperiment(win, 'bento-embedly-card', true, true);
+      // Override global window here because Preact uses global `createElement`.
+      doNotLoadExternalResourcesInTest(window, env.sandbox);
+    });
+
+    /**
+     * Wait for iframe to be mounted
+     */
+    const waitForRender = async () => {
+      await element.buildInternal();
+      const loadPromise = element.layoutCallback();
+      const shadow = element.shadowRoot;
+      await waitFor(() => shadow.querySelector('iframe'), 'iframe mounted');
+      await loadPromise;
+    };
+
+    it('renders responsively', async () => {
+      // Prepare Bento Tag
+      element = createElementWithAttributes(doc, 'amp-embedly-card', {
+        'data-url': 'https://www.youtube.com/watch?v=lBTCB7yLs8Y',
+        'height': 200,
+        'width': 300,
+        'layout': 'responsive',
+      });
+
+      // Add to Document
+      doc.body.appendChild(element);
+
+      // Wait till rendering is finished
+      await waitForRender();
+
+      //Extract iframe
+      const iframe = element.shadowRoot.querySelector('iframe');
+
+      //Make sure iframe is available
+      expect(iframe).to.not.be.null;
+
+      // Check iframe for correct scr URL
+      // <-- iframe.className remains '' -->
+      expect(iframe.className).to.match(/i-amphtml-layout-responsive/);
```

Comment:
The responsive class name is put on the element, not the inner iframe.
```suggestion
      expect(element.className).to.match(/i-amphtml-layout-responsive/);
```

You can also check that the iframe is styled with `width: 100%; height: 100%` or assert that the iframe and element have equal [`computedStyle`s](https://github.com/ampproject/amphtml/blob/2b0427db9b504e4983b01522d258e9c7f294da52/src/core/dom/style.js#L303-L306) for `width` and `height`.

Reply:
Is it correct to use as:
```
// Check that the iframe is styled with width: 100%; height: 100%
expect(iframe.style.width).to.equal('100%');
expect(iframe.style.height).to.equal('100%');

<-- This Works! -->
```

Following results with error:
```
expect(computedStyle(win, iframe).width).to.equal('100%');

<-- Error -->
AssertionError: expected '285px' to equal '100%'
```

Reply:
Instead of `expect(computedStyle(win, iframe).width).to.equal('100%');` you should be able to do `expect(computedStyle(win, iframe).width).to.equal(computedStyle(win, element).width);`